src/features/config/form/marriage/required-fields-marriage.ts

```diff
@@ -0,0 +1,264 @@
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
+ * graphic logo are (registered/a) trademark(s) of Plan International.
+ */
+
+import {
+  formMessageDescriptors,
+  informantMessageDescriptors
+} from '../formatjs-messages'
+import {
+  marriageDocumentForWhomFhirMapping,
+  marriageDocumentTypeFhirMapping
+} from '../options'
+import { ISelectOption, SerializedFormField, Conditional } from '../types/types'
+
+export const marriageInformantType: SerializedFormField = {
+  name: 'informantType',
+  type: 'SELECT_WITH_OPTIONS',
+  label: informantMessageDescriptors.birthInformantTitle,
+  required: true,
+  hideInPreview: false,
+  initialValue: '',
+  validator: [],
+  placeholder: formMessageDescriptors.formSelectPlaceholder,
+  mapping: {
+    mutation: {
+      operation: 'sectionFieldToBundleFieldTransformer',
+      parameters: ['registration.informantType']
+    },
+    query: {
+      operation: 'bundleFieldToSectionFieldTransformer',
+      parameters: ['registration.informantType']
+    },
+    template: {
+      fieldName: 'informantType',
+      operation: 'selectTransformer'
+    }
+  },
+  options: [
+    {
+      value: 'GROOM',
+      label: informantMessageDescriptors.GROOM
+    },
+    {
+      value: 'BRIDE',
+      label: informantMessageDescriptors.BRIDE
+    },
+    {
+      value: 'HEAD_OF_GROOM_FAMILY',
+      label: formMessageDescriptors.headOfGroomFamily
+    },
+    {
+      value: 'HEAD_OF_BRIDE_FAMILY',
+      label: formMessageDescriptors.headOfBrideFamily
+    },
+    {
+      value: 'OTHER',
+      label: informantMessageDescriptors.OTHER
+    }
+  ]
+}
+
+export const brideOrGroomBirthDateValidators = (spouseType: string) => [
+  {
+    operation: 'dateFormatIsCorrect',
+    parameters: []
+  },
+  {
+    operation: 'dateInPast',
+    parameters: []
+  },
+  {
+    operation: 'isValidDateOfBirthForMarriage',
+    parameters: [spouseType, 18]
+  }
+]
+
+export const getMarriageDate: SerializedFormField = {
+  name: 'marriageDate',
+  type: 'DATE',
+  label: formMessageDescriptors.marriageEventDate,
+  required: true,
+  initialValue: '',
+  validator: [
+    {
+      operation: 'checkMarriageDate',
+      parameters: [18]
+    }
+  ],
+  mapping: {
+    template: {
+      operation: 'marriageDateFormatTransformation',
+      fieldName: 'eventDate',
+      parameters: ['en', 'do MMMM yyyy', ['bride', 'groom']]
+    },
+    mutation: {
+      operation: 'fieldToMarriageDateTransformation',
+      parameters: [
+        ['bride', 'groom'],
+        {
+          operation: 'longDateTransformer',
+          parameters: []
+        }
+      ]
+    },
+    query: {
+      operation: 'marriageDateToFieldTransformation',
+      parameters: [['bride', 'groom']]
+    }
+  }
+}
+
+export const witnessName = (
+  name: string,
```

Comment:
```suggestion
  name: 'firstNamesEng' | 'familyNameEng',
```
using this should enable us to get rid of the ts-ignore